src/workers/test-explorer/logger.ts

```diff
@@ -1,7 +1,115 @@
+import { OutputChannel, window } from "vscode";
+import { TestResult } from "../../model/test-status.enum";
+
+export const OUTPUT_CHANNEL = "Angular Test Explorer";
+
+export enum LogLevel {
+  ERROR = "error",
+  WARN = "warn",
+  INFO = "info",
+  DEBUG = "debug",
+}
+
+interface Ilog {
+  message: string;
+  date: Date;
+  level: LogLevel;
+}
+
 export class Logger {
-  constructor() {}
+  private static outputChannel: OutputChannel | undefined;
+
+  public static setOutput() {
+    this.outputChannel = this.outputChannel || window.createOutputChannel(OUTPUT_CHANNEL);
+  }
+
+  public static debug(msg: string, ...params: any[]) {
+    const log: Ilog = {
+      message: msg,
+      date: new Date(),
+      level: LogLevel.DEBUG,
+    };
+
+    global.console.log(this.formatMsg(log));
+
+    if (this.outputChannel !== undefined) {
+      this.outputChannel.appendLine(this.formatMsg(log));
+    }
+  }
+
+  public static info(msg: string, ...params: any[]) {
+    const log: Ilog = {
+      message: msg,
+      date: new Date(),
+      level: LogLevel.INFO,
+    };
+
+    global.console.log(this.formatMsg(log));
+
+    if (this.outputChannel !== undefined) {
+      this.outputChannel.appendLine(this.formatMsg(log));
+    }
+  }
+
+  public static warn(msg: string, ...params: any[]) {
+    const log: Ilog = {
+      message: msg,
+      date: new Date(),
+      level: LogLevel.WARN,
+    };
+
+    global.console.log(this.formatMsg(log));
+
+    if (this.outputChannel !== undefined) {
+      this.outputChannel.appendLine(this.formatMsg(log));
+    }
+  }
+
+  public static error(msg: string, ...params: any[]) {
+    const log: Ilog = {
+      message: msg,
+      date: new Date(),
+      level: LogLevel.ERROR,
+    };
+
+    global.console.log(this.formatMsg(log));
+
+    if (this.outputChannel !== undefined) {
+      this.outputChannel.appendLine(this.formatMsg(log));
+    }
+  }
+
+  public static karmaLogsDivider() {
```

Comment:
can you make this private? try to think of a way that the logger use this when needed. I don't see a reason why the objects using the logger should decide or even know when the logger is dividing the content, this behavior should be transparent to them, also please give them a more descriptive name like divideKarmaLogsContent or something like that, always prefer longer and more descriptive methods over short and vague.